src/pkg/services/m365/api/channels_pager.go

```diff
@@ -0,0 +1,41 @@
+package api
```

Comment:
@neha-Gupta1 This file, and groups/handlers.go, were taken directly from [your PR here](https://github.com/alcionai/corso/pull/4050).  Had to copy them over because Abin's PR isn't based on them.

There's also some changes to that file here. Your PR is approved, let's get that pushed to main today.  After abin rebases/merges his branch, you'll be able to see the changes I made here.

Reply:
@ryanfkeepers there were some small changes which I have addressed in my PR. It would be good if my PR goes first and then this one. And once both of them are merged I will start implementing the missing pieces left.

Reply:
done